File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/character/CharacterHexaMatrixDTO.java
package dev.spiralmoon.maplestory.api.dto.character;

import com.google.gson.annotations.SerializedName;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 캐릭터 HEXA 코어 정보
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterHexaMatrixDTO {

    /**
     * 조회 기준일 (KST)
     */
    @SerializedName("date")
    private String date;

    /**
     * HEXA 코어 정보
     */
    @SerializedName("character_hexa_core_equipment")
    private List<CharacterHexaMatrixEquipmentDTO> characterHexaCoreEquipment;

    public LocalDateTime getDate() {
        return date != null
                ? Utils.toLocalDateTime(this.date)
                : null;
    }
}

File: js/src/maplestory/api/dto/character/characterHexaMatrixDto.ts
import { CharacterHexaMatrixEquipmentDto } from './characterHexaMatrixEquipmentDto';
import { CharacterHexaMatrixBody } from '../../response/character/characterHexaMatrixBody';

/**
 * 캐릭터 HEXA 코어 정보
 */
class CharacterHexaMatrixDto {
  /**
   * 조회 기준일
   */
  date: Date | null;

  /**
   * HEXA 코어 정보
   */
  characterHexaCoreEquipment: CharacterHexaMatrixEquipmentDto[];

  constructor(obj: CharacterHexaMatrixBody) {
    const { date, character_hexa_core_equipment } = obj;

    this.date = date ? new Date(date) : null;
    this.characterHexaCoreEquipment = character_hexa_core_equipment?.map(
      (equipment) => new CharacterHexaMatrixEquipmentDto(equipment),
    ) ?? [];
  }
}

export { CharacterHexaMatrixDto };
